### PR TITLE
Feature/interface clarifications

### PR DESCRIPTION
**Pull request by BeritJanssen**
This branch adds tooltips, and loads of `i18n` tags. I also added the `i18next-parser` to the package.json, and started writing a gulp script to compile messages. That's where I got stuck. @jgonggrijp : this is the branch I mentioned that might help with requesting translations soon.

### CODE REVIEW

**Comment by jgonggrijp** [comment]
@BeritJanssen I suspect I'm more or less done with this branch, although I'm a bit fuzzy on what was its original goal. Please help me decide!

I have done the following:

- Create infrastructure to automatically extract all the translation strings with their keys from the source code. Command to run: `yarn localize`.
- Reorganize the translation keys by topic and context (after discussing this with @JeltevanBoheemen) and uniformly use kebab-case for keys (because the underscore has special meaning to i18next, and camelCase is potentially less readable for translators).
- Identify a few remaining places where user-visible text needed to be i18n-ified.
- Generalize `TooltipView` so it can also be attached to subelements of other views and can also work with a model other than a `FlatItem`. I factored out the logic that extracts `skos:definition` or `rdfs:comment` from a `FlatItem` to a separate `TooltipModel` adapter. `TooltipView` now expects a model with just a `text` attribute, which is still expected to arrive async due to i18next.
- Add helper functions for adding tooltips to a view in bulk.
- Move some tooltips from template-based (with `data-tooltip`) to view-based (with `TooltipView` and its associated helper functions) in order to be able to position them more freely and comfortably.
- Expand a bit on some tooltips to give more guidance to users.
- Add a few more tooltips to the menu (which has changed since you started this branch) and to the `AnnotationPanel`.

I have **not** done the following (but maybe I should?):

- Exhaustively look for opportunities to add tooltips or other clarifications.

Since translatable strings are everywhere, changes are also everywhere. If you would like to look at the code, the most interesting changes are in the `frontend/src/{i18n,tooltip}` directories, the new `frontend/i18next-parser-config.mjs`, and `frontend/src/panel-annotation/annotation-{view.ts,template.hbs}`.

File: frontend/src/explorer/explorer-event-controller.ts [hunk 1]
```diff
@@ -311,10 +311,15 @@ class ExplorerEventController {
...
+        const i18nKey = `button.${title.toLowerCase().replace(' ', '-')}`;
         const endpoint = `${queryMode}:${landing ? 'user' : 'sample'}`;
         const collection = new FlatItemCollection(ldChannel.request(endpoint));
         const browsePanel = new SearchResultListPanel({
-            title,
+            // i18next.t('button.my-sources', 'My sources');
+            // i18next.t('button.sample-sources', 'Sample sources');
+            // i18next.t('button.my-items', 'My items');
+            // i18next.t('button.sample-items', 'Sample items');
+            title: i18next.t(i18nKey, title),
```
Comment (BeritJanssen) [inline frontend/src/explorer/explorer-event-controller.ts:322]: comments should probably go?

File: frontend/src/footer/footer-template.hbs [hunk 2]
```diff
@@ -1,5 +1,5 @@
 <figure class="footer-dh">
-    <p>Developed by</p>
+    <p>{{#i18n 'footer.developed-by'}}Developed by{{/i18n}}</p>
     <a href="https://dig.hum.uu.nl" target="_blank"><img class="dhlab"
             src="{{static 'image/dighum-logo-blue.svg'}}"></a>
```
Comment (BeritJanssen) [inline frontend/src/footer/footer-template.hbs:4]: Should we already proactively change the Footer to only include the UU logo and DHLab name?

File: frontend/src/i18n/en/translation_old.json [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    },
+    "source": {
+        "delete-confirm": "\nAre you sure you want to delete this source?\nIf you delete this source, all its annotation will be deleted as well, including any annotations that other users may have made.\nThis cannot be undone.\n"
+    },
+    "register": {
+        "thanks": {
+            "begin": "Thank you for signing up. You can now "
+        }
+    }
+}
```
Comment (BeritJanssen) [inline frontend/src/i18n/en/translation_old.json:48]: Do we need this file?

File: frontend/src/item-edit/detected-type-help-view.ts [hunk 4]
```diff
@@ -8,26 +8,14 @@ import { xsd } from '../common-rdf/ns';
...
-        'a derived string type.'
-    ),
-};
-
-const getHint = propertyOf(disambiguationHints);
+// i18next.t('property.typeHints.integer')
+// i18next.t('property.typeHints.base64Binary')
+// i18next.t('property.typeHints.gYear')
+// i18next.t('property.typeHints.string')
+const typesWithHints = ['integer', 'base64Binary', 'gYear', 'string'];
```
Comment (BeritJanssen) [inline frontend/src/item-edit/detected-type-help-view.ts:15]: Stray comments?

File: frontend/src/explorer/explorer-event-controller.ts (hunk 1 above)
Comment (jgonggrijp) [reply frontend/src/explorer/explorer-event-controller.ts:322]: No, they are needed to ensure that i18next-parser doesn't purge these keys on the next scan.

File: frontend/src/i18n/en/translation_old.json (hunk 3 above)
Comment (jgonggrijp) [reply frontend/src/i18n/en/translation_old.json:48]: Yes, it prevents the contained strings from becoming empty in the non-old version.

File: frontend/src/explorer/explorer-event-controller.ts (hunk 1 above)
Comment (JeltevanBoheemen) [reply frontend/src/explorer/explorer-event-controller.ts:322]: Can you elaborate on this? Why is there a difference in behaviour between commented code and removed code?

Comment (jgonggrijp) [reply frontend/src/explorer/explorer-event-controller.ts:322]: It's not outcommented. It's a "clarification" (for the parser) of what the next line means.

**Pull request merged**